schema/2.0.0/corpora_schema.md

```diff
@@ -0,0 +1,257 @@
+[mondo.owl]: https://github.com/monarch-initiative/mondo/releases/download/v2021-06-01/mondo.owl
+
+[Mouse Developmental Stages]: http://obofoundry.org/ontology/mmusdv.html
+[mmusdv.owl]: https://github.com/obophenotype/developmental-stage-ontologies/blob/0.1/src/mmusdv/mmusdv.owl
+
+[NCBI organismal classification]: http://obofoundry.org/ontology/ncbitaxon.html
+[2021-02-15]: https://github.com/obophenotype/ncbitaxon/releases/tag/v2021-02-15
+[ncbitaxon.owl]: https://github.com/obophenotype/ncbitaxon/releases/download/v2021-02-15/ncbitaxon.owl.gz
+
+[Phenotype And Trait Ontology]: http://www.obofoundry.org/ontology/pato.html
+[2021-05-26]: https://github.com/pato-ontology/pato/releases/tag/v2021-05-26
+[pato.owl]: https://github.com/pato-ontology/pato/blob/v2021-05-26/pato.owl
+
+[Uberon multi-species anatomy ontology]: http://www.obofoundry.org/ontology/uberon.html
+[2021-02-12]: https://github.com/obophenotype/uberon/releases/tag/v2021-02-12
+[uberon.owl]: https://github.com/obophenotype/uberon/blob/v2021-02-12/uberon.owl
+
+### Required Gene Annotations
+
+cellxgene requires ENSEMBL gene identifiers to ensure that all datasets it stores measure the same features and can therefore be integrated.
+
+The following gene annotation dependencies are *pinned* for this version of the schema.
+
+| Organism | Required version | Download |
+|:--|:--|:--|
+| [ENSEMBL (Human)] | Human reference GRCh38 (GENCODE v32/Ensembl 98)] matching [cellranger 2020-A (July 7, 2020) release] | [gencode.v32.primary_assembly.annotation.gtf] |
+| [ENSEMBL (Mouse)] | Mouse reference mm10 (GENCODE vM23/Ensembl 98) matching [cellranger 2020-A (July 7, 2020) release]| [gencode.vM23.primary_assembly.annotation.gtf] |
+|||
+
+[ENSEMBL (Human)]: http://www.ensembl.org/Homo_sapiens/Info/Index
+[gencode.v32.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.primary_assembly.annotation.gtf.gz"
+
+[ENSEMBL (Mouse)]: http://www.ensembl.org/Mus_musculus/Info/Index
+[gencode.vM23.primary_assembly.annotation.gtf]:http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M23/gencode.vM23.primary_assembly.annotation.gtf.gz"
+
+[cellranger 2020-A (July 7, 2020) release]: https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build
+
+## `obs` (Cell Metadata)
+
+`obs` is a [`pandas.DataFrame`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html).
+
+Curators MUST annotate the following columns in the `obs` dataframe:
+
+| Key | Value | Annotator |
+|-|-|-|
+| assay_ontology_term_id | categorical with `str` categories. This MUST be a child of "experimental process". See [EFO:0002694](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0002694) for details. If there is not an exact match for the assay, *clarifying text* MAY be appended in parentheses such as `" (sci-plex)"`. | Curator |
```

Comment:
Are you interpreting _This MUST be a child of "experimental process"._ as a requirement for labels instead of EFO term identifiers? We added this yesterday as a clarification, although I was worried that it would prove confusing. Previously, it was _child of EFO:0002694_.

Or are you recommending a specific list of examples using EFO identifiers? For example, this is the current set of assays (with many duplicate labels at the moment):

```
{"10X 3' v2 sequencing : EFO:0009899",
 '10X sequencing : EFO:0008995',
 "10x 3' v2 : EFO:0009899",
 "10x 3' v2 sequencing : EFO:0009899",
 "10x 3' v3 : EFO:0009922",
 "10x 3' v3 sequencing : EFO:0009922",
 "10x 5' v1 : ",
 "10x 5' v3 sequencing : EFO:0009921",
 '10x Genomics : EFO:0008995',
 '10x technology : EFO:0008995',
 '10x v2 : EFO:0009310',
 '10x v3 : EFO:0009898',
 '10x v3 sequencing : EFO:0009898',
 'ATAC 10x v1 : ',
 'CITE-Seq : EFO:0009294',
 'CITE-seq : EFO:0009294',
 'Drop-seq : EFO:0008722',
 'MERFISH : EFO:0008992',
 'Seq-Well : EFO:0008919',
 'Smart-seq : EFO:0008930',
 'Smart-seq2 : EFO:0008931',
 'Smart-seq2 protocol : EFO:0008442',
 'Visium Spatial Gene Expression : EFO:0010961',
 'methylation profiling : EFO:0000751',
 'microwell-seq : ',
 'scATAC-seq : EFO:0010891',
 'scRNA-seq : EFO:0008913',
 'sci-RNA-seq : EFO:0010550',
 'sci-plex : '}
```

Reply:
The latter. I would suggest the latter, a specific list of examples.
Currently, the data from existing datasets of '10X sequencing', '10x 3' v3 sequencing', and 'scRNA-seq' sequencing are all produced from the same assay, yet their metadata do not reflect that even though each follows the schema requirements.
Some additional guidance or requirement, or list of examples, is needed to ensure that a single process is represented consistently across the submissions.

Reply:
> Currently, the data from existing datasets of '10X sequencing', '10x 3' v3 sequencing', and 'scRNA-seq' sequencing **are all produced from the same assay,** yet their metadata do not reflect that even though each follows the schema requirements.

What is your specific proposal because I can interpret your remarks in two ways:

1. Require more specific terms than '10X sequencing' and 'scRNA-seq'. (And I note that the '10X sequencing label' does not match its EFO identifier.)
2. Use the most high-level term for a family of assays such as [EFO:0008995](https://www.ebi.ac.uk/ols/ontologies/efo/terms?short_form=EFO_0008995) for
'10x technology' which would not align with our guidance to use the most specific term available.

Reply:
Door `1`, please - require more specific terms
I think the guidance to use the most specific term available is not very clear.
I see "When no appropriate ontology value is available, then the most accurate term MUST be used.", which is close, but not as direct. Perhaps rephrasing that to "The most accurate term MUST be used, even when no available ontology term perfectly captures the value"...but then note the exceptions (development stage)

Reply:
@ambrosejcarr, @pablo-gar, and I need to review how to _tighten up_ the guidance for assays. Perhaps, if it's 10X technology, then it MUST be a child of EFO:0008995. We need a MUST statement that can be implemented in code; otherwise, it's a SHOULD.

Reply:
>We need a MUST statement that can be implemented in code; otherwise, it's a SHOULD.

I think it's going to be a SHOULD. You won't know that it's 10x unless you're reading the paper, so the code won't know if 'scRNA-seq' is an allowable term.

Reply:
Reviewing a spreadsheet of our current set of assays and their "parents" with Ambrose and Pablo. There are already some obsolete terms too.

Reply:
Something like:

An assay based on 10X Genomics products SHOULD be EFO:0008995 for “10x technology” or the most accurate child of EFO:0008995. Other assays SHOULD be the most accurate child of either EFO:0002772 for “assay by molecule” or EFO:0010183  for “single cell library construction”. If there is not an exact match for the assay, *clarifying text* MAY be appended to the most accurate term in parentheses. For example, the sci-plex assay would be curated as “EFO:0010183 (sci-plex)” and cellxgene Data Portal would set its name to “single cell library construction (sci-plex)“.

Reply:
<table><tbody>
    <tr>
      <th>Key</th>
      <td>assay_ontology_term_id</td>
    </tr>
    <tr>
      <th>Value</th>
        <td>This and that and the other thing. on and on and on. Recommended values per assay:
          <ul>
          <li>Use EFO00000 for 10x technology </li>
          <li>Use EFO00001 for SmartSeq-2 </li>
          <li>Use EFO00002 for ATAC-seq </li>
          </ul>
        </td>
    </tr>
    <tr>
    <th>Annotator</th>
    <td>Curator<td>
    </tr>
</tbody></table>

Reply:
Addressed in https://github.com/chanzuckerberg/single-cell-curation/pull/67/commits/138a06464458673c98ba4d2c4618f8eb4e5be477